File: src/java/org/javaturk/dp/ch02/elevator/EncapsulatedElevatorTest.java
package org.javaturk.dp.ch02.elevator;

import org.javaturk.dp.ch02.elevator.domain.EncapsulatedElevator;
import org.javaturk.dp.ch02.elevator.domain.Floor;

public class EncapsulatedElevatorTest {

	public static void main(String args[]) {

		EncapsulatedElevator encapElevator = new EncapsulatedElevator();

		encapElevator.openDoor(); // Get into the elevator
		encapElevator.goDown();
		encapElevator.goUp();
		encapElevator.goUp();
		encapElevator.closeDoor();
		encapElevator.openDoor(); // Somebody is coming :)
		encapElevator.goDown();
		encapElevator.goDown();
		encapElevator.goDown();
		encapElevator.go(Floor.FOURTH);
	}
}

File: src/c#/ch02/Elevator/EncapsulatedElevatorTest.cs
using System;
using Elevator.Domain;

namespace Elevator
{
    public class EncapsulatedElevatorTest
    {
        public static void Main(string[] args)
        {
            Console.WriteLine("EncapsulatedElevatorTest");

            EncapsulatedElevator encapElevator = new EncapsulatedElevator();

            encapElevator.OpenDoor(); // Get into the elevator
            encapElevator.GoDown();
            encapElevator.GoUp();
            encapElevator.GoUp();
            encapElevator.CloseDoor();
            encapElevator.OpenDoor(); // Somebody is coming :)
            encapElevator.GoDown();
            encapElevator.GoDown();
            encapElevator.GoDown();
            encapElevator.Go(Floor.FOURTH);
        }
    }
}
